src/main/java/baritone/process/MineProcess.java

```diff
@@ -78,6 +80,52 @@ public boolean isActive() {
 
     @Override
     public PathingCommand onTick(boolean calcFailed, boolean isSafeToCancel) {
+        if (Baritone.settings().miningCheckInventory.value) {
+           boolean inventoryFull = true;
+           NonNullList<ItemStack> invy = ctx.player().inventory.mainInventory;
+           List<Item> DropsHaveSpace = new ArrayList<Item>();
+           for (ItemStack stack:invy) {
+
+               if (stack.isEmpty()) {
+                   inventoryFull = false;
+                   break;
+               }
+               List<Item> miningDrops = new ArrayList<Item>();
+               for (Block block : mining) {
+
+                       miningDrops.add(block.getItemDropped(mining.get(0).getDefaultState(), new Random(), 0));
+               }
+
+               if (miningDrops.contains(stack.getItem()) && stack.getMaxStackSize() != stack.getCount()) {
+                   DropsHaveSpace.add(stack.getItem());
+               }
+
+           }
+           if (inventoryFull && !DropsHaveSpace.isEmpty()) {
+               inventoryFull = false;
+               for (Block block : mining) {
+
+
+                       if (!DropsHaveSpace.contains(block.getItemDropped(mining.get(0).getDefaultState(), new Random(), 0))) {
+                           mining.remove(block);
+                       }
+
+               }
+
+           }
+
+           if (inventoryFull) {
+               logDirect("Cancel Mining Inventory Full");
+               if (Baritone.settings().miningGoHome.value) {
+                   Goal goal = new GoalBlock(baritone.getWorldProvider().getCurrentWorld().getWaypoints().getMostRecentByTag(IWaypoint.Tag.HOME).getLocation());
```

Comment:
causes NPE when no recent home waypoint is found (lines 147, 157, 198 & 268 cause the same NPE)
```suggestion
                    IWaypoint waypoint = baritone.getWorldProvider().getCurrentWorld().getWaypoints().getMostRecentByTag(IWaypoint.Tag.HOME);
                    if (waypoint != null) {
                        Goal goal = new GoalBlock(waypoint.getLocation());
                        baritone.getCustomGoalProcess().setGoalAndPath(goal);
                    } else {
                        logDirect("No recent waypoint found, can't return home");
                    }
```